common/bin/download-verify-npm-package

```diff
@@ -19,28 +19,32 @@ if [ -z "$package_version" ]; then
 	exit 1
 fi
 
+package_name="${distribution_name}"
 if [ "yarn" = "${package_name}" ]; then
 	# Yarn 2+ (aka: "berry") is hosted under a different npm package.
 	major_version=$(echo "$package_version" | cut -d "." -f 1)
 	package_name=$([ "$major_version" -ge 2 ] && echo "@yarnpkg/cli-dist" || echo "yarn")
 fi
 
 npm_url="https://registry.npmjs.com/${package_name}/${package_version}"
-
 echo "Determining dist url from ${npm_url}" >&2
-url=$(curl -sSf "${npm_url}" | jq -r '.dist.tarball')
 
-echo "Downloading ${package_name} tarball from ${url} ..." >&2
-curl -sSf -o "./${package_name}-v${package_version}.tar.gz" "${url}"
+tarball_url=$(curl --silent --show-error --fail --retry 5 --retry-all-errors --connect-timeout 10 --max-time 60 "${npm_url}" | jq -r '.dist.tarball')
+echo "Downloading ${package_name} tarball from ${tarball_url} ..." >&2
+
+download_file="./${distribution_name}-v${package_version}.tar.gz"
```

Comment:
Nit, optional and non-blocking: `downloaded_file` seems to be a better name here.

Reply:
changed it to `downloaded_tarball`